Felienne/hedy
# Workaround for 'copy share link to clipboard' in Safari after sharing program

**Description**

Currently when making a program public from either the main programming page  or the programs page, we copy a share link to the clipboard. This, however, doesn't work in Safari because Safari doesn't allow to copy things to a clipboard after an API call if there hasn't been a click.

To solve this, we stop copying the share link altogether when making a program public, and instead add a 'copy share link' button on the modal that tells the user that the program has been saved successfully.

**Fix for**

- Closes #548 

**How to test**

- Share a program from the main programming view and make sure that there's a link to copy the program to the clipboard. Once you click on it, verify that a share link has indeed been copied to the clipboard.
- Repeat the above operation, but from the Programs page

**Implementation notes**

- it was also necessary to add extra logic to the Modal class so that if a modal B was shown before a modal A was dismissed, the timeout for hiding modal A would not hide modal B.
- We repurposed a text from the Programs section and moved it to the Auth section within the language yamls.
